File: solving/기타알고리즘/구현/01-달팽이 배열.py
import sys
getInput = sys.stdin.readline

n = int(getInput())
snail_array = [[0] * n for _ in range(n)]

def print_snail_array(snail_array):
  length = len(snail_array)
  for i in range(length):
    print(snail_array[i][:length])

def make_snail_array(n, snail_array):
  num = 0
  row = 0
  col = -1
  inc = 1

  while n > 0:

    for _ in range(n):
      col += inc
      num += 1
      snail_array[row][col] = num

    n -= 1

    for _ in range(n):
      row += inc
      num += 1
      snail_array[row][col] = num

    inc *= -1

make_snail_array(n, snail_array)

print_snail_array(snail_array)

File: solving/기타알고리즘/구현/01-달팽이 배열.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const makeSnailArray = (n, snail_array) => {
  let num = 0;
  let row = 0;
  let col = -1;
  let inc = 1;

  while (n > 0) {
    for (let i = 0; i < n; i++) {
      num++;
      col += inc;
      snail_array[row][col] = num;
    }

    n--;

    for (let j = 0; j < n; j++) {
      num++;
      row += inc;
      snail_array[row][col] = num;
    }

    inc *= -1;
  }
};

const solution = (line) => {
  const n = parseInt(line);
  const snail_array = Array.from({ length: n }, () => Array(n).fill(0));

  makeSnailArray(n, snail_array);

  console.log(snail_array);
  rl.close();
};

rl.on("line", solution).on("close", () => process.exit());
